web/src/rest/publish.js

```diff
@@ -89,16 +95,17 @@ const publishRest = new Vue({
       }
     },
     async mostRecentVersion(identifier) {
-      const versions = await this.versions(identifier);
-      if (versions === null) {
+      // TODO: find a way to do this with fewer requests
+      const count = (await this.versions(identifier))?.count;
+      if (!count) {
         return null;
       }
-      const { count, results } = versions;
-      if (count === 0) {
-        return null;
-      }
-      const { version } = results[0];
-      return this.specificVersion(identifier, version);
+      // Look up the last version using page filters
+      const version = (await this.versions(identifier, { page: count, page_size: 1 })).results[0];
```

Comment:
This wouldn't always apply (e.g. if there's many versions) but couldn't you do the following?

```
const versions = await this.versions(identifier);
if (!versions.count) { return null; }
if (!versions.next) {
  const sortFunc = (a, b) => (a.version > b.version ? -1 : 1);
  const latestVersion = versions.results.filter(
    (version) => version.version !== 'draft'
  ).sort(sortFunc)[0];
  return girderize(latestVersion);
}

// Default to existing implementation
```

I lightly tested this in my browser and it performed as expected.

Reply:
I did it this way specifically for the case when there are more versions than fit in a single page. My solution is the simplest one I could think of that always works, at the cost of some extra requests. Doing it your way would produce some confusing behavior at some point in the future when someone publishes one too many times.

Reply:
If a dandiset has more versions than fit on a page, the `next` field would not be `null`, and this block wouldn't run, correct?

I'm fine with not implementing this for now, just thought I'd offer a way to possibly save on requests.

Reply:
We decided to leave as is